RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -949,18 +983,32 @@ double AntiElectronIDMVA6::MVAValue(const pat::Tau& theTau, const pat::Electron&
   Int_t TauSignalPFGammaCandsIn = GammasPtInSigCone.size();
   Int_t TauSignalPFGammaCandsOut = GammasPtOutSigCone.size();
   Float_t TauVisMassIn = (pfGammaSum + pfChargedSum).mass();
-
-  Float_t TauPhi = theTau.phi();
- 
-  if ( theTau.phiAtEcalEntrance() != -99. ) {
-    TauPhi = theTau.phiAtEcalEntrance();
+  Float_t TauPhi = -99.;
+  if ( usePhiAtEcalEntranceExtrapolation ) {
+    float sumPhiTimesEnergy = 0.;
+    float sumEnergy = 0.;
+    const reco::CandidatePtrVector signalCands = theTau.signalCands();
+    for (unsigned int o = 0; o < signalCands.size(); o++ ) {
+      reco::Candidate const* signalCand = signalCands[o].get();
+      float phi = theTau.phi();
+      math::XYZPoint aPos;
+      if ( atECalEntrance(signalCand, aPos) == true ) phi = aPos.Phi();
```

Comment:
Hi @roger-wolf - just to be sure I follow - this phi variable ends up meaning "phi at ecal" if the particle intersects the ecal, otherwise it is phi at the primary vertex?